src/app/(sidebar)/xdr/view/page.tsx

```diff
@@ -137,10 +154,14 @@ export default function ViewXdr() {
             disabled={isFetchingLatestTxn}
           />
 
-          <TransactionHashReadOnlyField
-            xdr={xdr.blob}
-            networkPassphrase={network.passphrase}
-          />
+          {xdr.type?.includes("Transaction") ? (
+            <TransactionHashReadOnlyField
+              xdr={xdr.blob}
+              networkPassphrase={network.passphrase}
+            />
+          ) : (
+            <></>
+          )}
```

Comment:
Maybe instead of doing this check, we could handle the logic inside the `TransactionHashReadOnlyField` component. If there is no hash to show, we don't render the component, no matter what type it is.

```typescript
export const TransactionHashReadOnlyField = ({
  xdr,
  networkPassphrase,
}: {
  xdr: string;
  networkPassphrase: string;
}) => {
  if (!xdr) {
    return null;
  }

  const hashValue = transactionHashFromXdr(xdr, networkPassphrase);

  if (!hashValue) {
    return null;
  }

  return (
    <Input
      id="tx-hash"
      fieldSize="md"
      label="Transaction hash"
      value={hashValue}
      copyButton={{ position: "right" }}
      disabled
    />
  );
};
```